enhancements/microshift/low-latency-workloads-on-microshift.md

```diff
@@ -0,0 +1,612 @@
+
+
+### Drawbacks
+
+Approach described in this enhancement does not provide much of the NTO's functionality
+due to the "static" nature of RPMs and packaged files (compared to NTO's dynamic templating),
+but it must be noted that NTO is going beyond low latency.
+
+One of the NTO's strengths is that it can create systemd units for runtime configuration
+(such as offlining CPUs, setting hugepages per NUMA node, clearing IRQ balance banned CPUs,
+setting RPS masks). Such dynamic actions are beyond capabilities of static files shipped via RPM.
+If such features are required by users, we could ship such systemd units and they could be no-op
+unless they're turned on in MicroShift's config. However, it is unknown to author of the enhancement
+if these are integral part of the low latency.
+
+## Open Questions [optional]
+
+- Verify if osbuild blueprint can override a file from RPM 
+  (variables.conf needs to exist for tuned profile, so it's nice to have some fallback)?
+- NTO runs tuned in non-daemon one shot mode using systemd unit.
+  Should we try doing the same or we want the tuned daemon to run continuously?
+- NTO's profile includes several other beside cpu-partitioning: 
+  [openshift-node](https://github.com/redhat-performance/tuned/blob/master/profiles/openshift-node/tuned.conf)
+  and [openshift](https://github.com/redhat-performance/tuned/blob/master/profiles/openshift/tuned.conf) - should we include them or incorporate their settings?
+- NTO took an approach to duplicate many of the setting from included profiles - should we do the same?
+  > Comment: Probably no need to do that. `cpu-partitioning` profile is not changed very often,
+  > so the risk of breakage is low, but if they change something, we should get that automatically, right?
+- Should we also provide NTO's systemd units for offlining CPUs, setting hugepages per NUMA node, clearing IRQ balance, setting RPS masks?
+
+## Test Plan
+
+Two aspect of testing:
+- Configuration verification - making sure that what we ship configures what we need.
+  Previously mentioned `microshift doctor low-latency` might be reference point.
+
+- Runtime verification - making sure that performance is as expected.
+  This might include using tools such as: hwlatdetect, oslat, cyclictest, rteval, and others.
+  Some of the mentioned tools are already included in the `openshift-tests`.
+  This step is highly dependent on the hardware, so we might need to long-term lease some hardware in
+  Beaker to have consistent environment and results that can be compared between runs.
+
+## Graduation Criteria
+
+Feature is meant to be GA on first release.
+
+### Dev Preview -> Tech Preview
+
+Not applicable.
+
+### Tech Preview -> GA
+
+Not applicable.
+
+### Removing a deprecated feature
+
+Not applicable.
+
+## Upgrade / Downgrade Strategy
+
+Upgrade / downgrade strategy is not needed because there are almost no runtime components or configs
+that would need migration.
```

Comment:
We are adding a new section to config.yaml. How would a previous version of MicroShift behave if downgraded?

Reply:
I'll double check but probably it'll be ignored - stuff not present in the Golang structure will just be ignored